src/audio/copier/copier.c

```diff
@@ -1491,6 +1491,13 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
```

Comment:
”Fixes audio distortion when host gateway is used in capture mode with different container size for its source and sink.”

Again, host gateway case should not enter this condition, it should go to line 1508:

				ret = cd->endpoint[i]->drv->ops.params(cd->endpoint[i],
								       params);